broker-core/src/test/java/io/zeebe/broker/workflow/timer/TimerCatchEventTest.java

```diff
@@ -357,45 +358,46 @@ public void shouldCancelTimer() {
             .withWorkflowInstanceKey(workflowInstanceKey)
             .withHandlerNodeId("timer")
             .getFirst();
+
     testClient.cancelWorkflowInstance(workflowInstanceKey);
 
     // then
     final Record<TimerRecordValue> canceledEvent =
         RecordingExporter.timerRecords(TimerIntent.CANCELED)
             .withWorkflowInstanceKey(workflowInstanceKey)
-            .withHandlerNodeId("timer")
             .getFirst();
+
     assertThat(canceledEvent.getKey()).isEqualTo(createdEvent.getKey());
     assertThat(canceledEvent.getValue()).isEqualTo(createdEvent.getValue());
-    assertThat(canceledEvent.getValue().getDueDate())
-        .isGreaterThan(brokerRule.getClock().getCurrentTimeInMillis());
   }
 
   @Test
   public void shouldCreateTimerBasedOnBoundaryEvent() {
-    // given
+    // given/when
     final long workflowKey = testClient.deployWorkflow(BOUNDARY_EVENT_WORKFLOW).getKey();
-    brokerRule.getClock().pinCurrentTime();
-    final long nowMs = brokerRule.getClock().getCurrentTimeInMillis();
     final long workflowInstanceKey =
         testClient.createWorkflowInstance(r -> r.setKey(workflowKey)).getInstanceKey();
 
-    // when
-    final Record<TimerRecordValue> timerCreatedRecord =
+    // then
+    final Record<TimerRecordValue> timerRecord =
         RecordingExporter.timerRecords(TimerIntent.CREATED)
             .withWorkflowInstanceKey(workflowInstanceKey)
             .getFirst();
+
     final Record<WorkflowInstanceRecordValue> activityRecord =
         RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_ACTIVATED)
             .withWorkflowInstanceKey(workflowInstanceKey)
             .withElementId("task")
             .getFirst();
 
-    // then
-    assertThat(timerCreatedRecord.getValue().getDueDate()).isEqualTo(nowMs + 1000);
-    assertThat(timerCreatedRecord.getValue().getElementInstanceKey())
-        .isEqualTo(activityRecord.getKey());
-    assertThat(timerCreatedRecord.getValue().getHandlerFlowNodeId()).isEqualTo("timer");
+    Assertions.assertThat(timerRecord.getValue())
+        .hasElementInstanceKey(activityRecord.getKey())
+        .hasHandlerFlowNodeId("timer");
+
+    assertThat(timerRecord.getValue().getDueDate())
```

Comment:
It could still happen that If we wait on one of the events with the Recording exporter longer than 1 second, our lower bound is not longer correct right?